lib/errors.js

```diff
@@ -0,0 +1,74 @@
+class StorageEmpty extends Error {
+  constructor () {
+    super('No Hypercore is stored here')
+    this.code = 'STORAGE_EMPTY'
+  }
+}
+
+class StorageConflict extends Error {
+  constructor () {
+    super('Another Hypercore is stored here')
+    this.code = 'STORAGE_CONFLICT'
+  }
+}
+
+class InvalidSignature extends Error {
+  constructor () {
+    super('Remote sent invalid signature')
+    this.code = 'INVALID_SIGNATURE'
+  }
+}
+
+class InvalidCapability extends Error {
+  constructor () {
+    super('Remote sent invalid capability')
+    this.code = 'INVALID_CAPABILITY'
+  }
+}
+
+class SnapshotNotAvailable extends Error {
+  constructor () {
+    super('Snapshot is not available')
+    this.code = 'NOT_AVAILABLE'
+  }
+}
+
+class RequestCancelled extends Error {
+  constructor () {
+    super('Request cancelled')
+    this.code = 'CANCELLED'
```

Comment:
Same as above comment, `REQUEST_CANCELLED` vs `CANCELLED`

Reply:
Same reason, generic code for a cancelled action.

Reply:
This one prob makes more sense than the one above. Could also be ABORTED as a generic Abort error (ie when we add some abort signals etc).

Reply:
Makes sense to just be `CancellationError` or `Cancel` then to keep it general